usagov/vote-gov-drupal
# Feature/VOTE-965: Tome generation improvements

## Jira ticket

[VOTE-965](cm-jira.usa.gov/browse/VOTE-965)

## Description

Update static site generation script to clean up unwanted files, improve processing, and replace absolute asset links for uploaded files.

## Deployment and testing

### Pre-deploy

n/a

### Post-deploy

1. Await for static build pipeline to complete.

### QA/Test

1. Once static site pipeline is complete confirm that the following paths are not accessible from the static site:
- /taxonomy/term/{tid}(numeric number)
- /node/{nid} (67 - 70) - try any of those numbers
- From the home page, locate the hero image and confirm that the image url is relative (/s3/files/...)

Local Testing:
1. Run `lando retune` to rebuild site locally.
2. Run `lando static` to build your static site locally.
3. From the command line, `cd html` from the project root.
4. Confirm that the taxonomy, node, and {langcode}/node directories exist.
5. Run `find . -type d -name "node" -print -exec rm -rf {} +`
6. Confirm that the node directories are deleted.
7. Run `rm -rf taxonomy'
8. Confirm that the taxonomy directory is deleted.
9. Locate a file with an asset/image url that includes 's3/files' or create an entry that follows the url pattern "https://cms-ssg.vote.gov/s3/files/....../*.png|jpg|jpeg".
10. Run `find . -type f -exec grep -l 'http[s]*://[^/]\+/s3/files' {} \;` to confirm that the filename is listed in the terminal.
11. Run `find . -type f -exec grep -l 'http[s]*://[^/]\+/s3/files' {} \; -exec sed -i 's#http[s]*://[^/]\+/s3/files#/s3/files#g' {} +`
12. Confirm that the url in the file is now a relative path.

Cloud Environment Testing:
1. Visit static site for a given environment (ie. ssg-test.vote.gov).
2. Inspect the homepage and confirm the following:
-- No missing assets are reported or CORS alerts (images, fonts, etc.)
-- Inspect the hero image and confirm that the url is a relative url
-- Using the 'Sources' tab in the browser inspection tool, locate an aggregated css file /sites/default/files/css/css_.... and confirm that asset paths are relative urls (ie. background-image: url(/themes/...))
3. Confirm that /sitemap.xml urls reflect the correct domain (once deployed to prod, it should be beta.vote.org)
4. Confirm that taxonomy paths are unavailable (ie. /taxonomy/term/41)
5. Confirm that node/# paths which have a url-alias is not available (ie. /node/3 == /about-us)

## Checklist for the Developer




## Checklist for the Peer Reviewers
